source/game.c

```diff
@@ -2767,39 +2761,33 @@ static int game_shop_get_random_joker_idx()
 static void game_shop_create_items()
 {
     tte_erase_rect_wrapper(SHOP_PRICES_TEXT_RECT);
-    if (list_get_size(jokers_available_to_shop) == 0)
-    {
-        // No jokers to create
-        return;
-    }
 
-    shop_jokers = list_new(MAX_SHOP_JOKERS);
+    if (_no_avail_jokers()) return;
+
+    list_destroy(&_shop_jokers_list);
+    _shop_jokers_list = list_new();
 
     for (int i = 0; i < MAX_SHOP_JOKERS; i++)
     {
         intptr_t joker_id = 0;
```

Comment:
```suggestion
        int joker_id = 0;
```

I guess this should be changed from inptr_t now as there's no reason for it